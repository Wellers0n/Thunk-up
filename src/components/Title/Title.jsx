import React from "react";
import PropTypes from "prop-types";
import Typography from "@material-ui/core/Typography";
import useStyles from './Style'

export default function Title(props) {
  const classes = useStyles();


  return (
    <Typography
      className={classes.root}
      align="center"
      component="h2"
      variant="h6"
      gutterBottom
    >
      <div data-testid="nome">{`${props.nome}'s Challenge`}</div>
    </Typography>
  );
}

Title.propTypes = {
  children: PropTypes.node,
  nome: PropTypes.string
};

Title.defaultProps = {
  nome: "Define name"
}
