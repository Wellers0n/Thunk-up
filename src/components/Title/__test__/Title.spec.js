import React from "react";
import { render, waitForElement, fireEvent } from "@testing-library/react";
import Title from "../Title";

describe("App tests", () => {
  it("should render Title with name Wellerson", async () => {
    const { getByTestId, getByText } = render(<Title nome="Wellerson" />);
    const nome = await waitForElement(() => getByTestId("nome"));
    expect(nome.textContent).toBe("Wellerson's Challenge");
  });

  it("should render Title with name DK", async () => {
    const { getByTestId, getByText } = render(<Title nome="DK" />);

    const nome = await waitForElement(() => getByTestId("nome"));
    expect(nome.textContent).toBe("DK's Challenge");
  });

  it("snapshot of Title", () => {
    const { container } = render(<Title nome="nice" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render Title without name === Define name", async () => {
    const { getByTestId, getByText } = render(<Title />);

    const nome = await waitForElement(() => getByTestId("nome"));
    expect(nome.textContent).toBe("Define name's Challenge");
  });
});