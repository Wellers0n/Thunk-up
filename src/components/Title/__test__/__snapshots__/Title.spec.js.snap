// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Title snapshot of Title 1`] = `
<h2
  class="MuiTypography-root makeStyles-root-63 MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter"
>
  <div
    data-testid="nome"
  >
    nice's Challenge
  </div>
</h2>
`;
